src/legacy/btchip_apdu_hash_input_finalize_full.c

```diff
@@ -70,12 +71,17 @@ static bool check_output_displayable() {
     isOpCall =
         btchip_output_script_is_op_call(btchip_context_D.currentOutput + 8,
           sizeof(btchip_context_D.currentOutput) - 8);
-    if (((G_coin_config->kind == COIN_KIND_QTUM || G_coin_config->kind == COIN_KIND_HYDRA) &&
-         !btchip_output_script_is_regular(btchip_context_D.currentOutput + 8) &&
-         !isP2sh && !(nullAmount && isOpReturn) && !isOpCreate && !isOpCall) ||
-        (!(G_coin_config->kind == COIN_KIND_QTUM || G_coin_config->kind == COIN_KIND_HYDRA) &&
-         !btchip_output_script_is_regular(btchip_context_D.currentOutput + 8) &&
-         !isP2sh && !(nullAmount && isOpReturn))) {
+    if (btchip_output_script_is_regular(btchip_context_D.currentOutput + 8) ||
+        isP2sh || (nullAmount && isOpReturn)) {
+        isRecognizedOutputScript = true;
+    } else if (G_coin_config->kind == COIN_KIND_QTUM || G_coin_config->kind == COIN_KIND_HYDRA) {
+        if (isOpCreate || isOpCall)
+            isRecognizedOutputScript = true;
+    } else if (G_coin_config->kind == COIN_KIND_PEERCOIN) {
+        if (nullAmount)
+            isRecognizedOutputScript = true;
```

Comment:
Alright, the change now seems ok with respect to the other coins.

Regarding Peercoin, this is saying that any script output with null amount is now considered valid. Is this intended behavior? Does it really need to be completely unchecked, or can a condition be added with the right script type?
What happens when a random/invalid script is passed? What is shown on the screen for that output, or does the app crash?

Reply:
Great feedback, I've added extra condition to check for empty script, because coinstake output must have empty script. 

Another thing I've noticed, is that current logic (and old one as well) allows for nullAmount if output script considered regular. Should we add extra check with && !nullAmount to first if?

Reply:
I think that's fine, as a zero-value output does not pose any security risk.
The general idea is to whitelist specific supported valid scripts, and only those.

Last question, otherwise I think it's good: have you checked that the app shows something meaningful for those outputs? Maybe something needs to be added in `main_old.c::get_address_from_output_script`?

Reply:
Thanks for the tip, it now looks much better.